fdbserver/workloads/TenantCapacityLimits.actor.cpp

```diff
@@ -76,38 +76,23 @@ struct TenantCapacityLimits : TestWorkload {
 	}
 	ACTOR static Future<Void> _setup(Database cx, TenantCapacityLimits* self) {
 		if (self->useMetacluster) {
-			Reference<IDatabase> threadSafeHandle =
-			    wait(unsafeThreadFutureToFuture(ThreadSafeDatabase::createFromExistingDatabase(cx)));
-
-			MultiVersionApi::api->selectApiVersion(cx->apiVersion.version());
-			self->managementDb = MultiVersionDatabase::debugCreateFromExistingDatabase(threadSafeHandle);
-
-			wait(success(metacluster::createMetacluster(
-			    cx.getReference(), "management_cluster"_sr, self->tenantIdPrefix, false)));
+			ASSERT(g_simulator->extraDatabases.size() == 1);
 
 			metacluster::DataClusterEntry entry;
 			entry.capacity.numTenantGroups = 1e9;
-			wait(metacluster::registerCluster(
-			    self->managementDb, "test_data_cluster"_sr, g_simulator->extraDatabases[0], entry));
 
-			ASSERT(g_simulator->extraDatabases.size() == 1);
-			self->dataDb = Database::createSimulatedExtraDatabase(g_simulator->extraDatabases[0], cx->defaultTenant);
-			// wait for tenant mode change on dataDB
-			wait(success(self->waitDataDbTenantModeChange()));
+			metacluster::util::SimulatedMetacluster simMetacluster =
+			    wait(metacluster::util::createSimulatedMetacluster(cx, self->tenantIdPrefix, entry));
+
+			self->managementDb = simMetacluster.managementDb;
+			self->dataDb = simMetacluster.dataDbs.begin()->second;
```

Comment:
I wonder whether it's useful to add a check for the size of `simMetacluster.dataDbs`

Reply:
We could, though part of the goal of this was to minimize the boilerplate that has to be written to set up a metacluster. One of the guarantees of the function is that there will be a dataDb for every item in extraDatabases, and we have already asserted that there is an extra database above. Perhaps we could eliminate the assert at line 79 and replace it with an assert about the number of dataDbs?